code/modules/modular_computers/computers/item/computer.dm

```diff
@@ -487,7 +487,7 @@
 
 /obj/item/modular_computer/screwdriver_act(mob/user, obj/item/tool)
 	if(!length(all_components))
-		to_chat(user, span_warning("This device doesn't have any components installed."))
+		user.balloon_alert(user, "this device doesn't have any components installed")
```

Comment:
```suggestion
		user.balloon_alert(user, "no components installed!")
```

Balloon alerts must be short, and use `!` to designate failure.

Reply:
Balloon alerts should also source from the object itself, because that is where you are already looking.

Reply:
So I don't need to source from user? Should there be anything else in place of that or is it fine to just use balloon_alert

Reply:
`balloon_alert` is an atom level proc, it's where the balloon alert comes from.

You use `user.balloon_alert` if it relates to the user, or if the object is not going to exist by the time you need to balloon alert (such as deconstruction).